### PR TITLE
RD-5769 Migrate Events Filter widget

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
## Description

RD-5769
- migrate Events Filter widget to TS
- migrate translation to `en.json`

## Screenshots / Videos

n/a

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

there is no test for Events Filter Widget

## Documentation

n/a

### CODE REVIEW

File: app/widgets/common/utils/EventUtils.ts [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+type EventTypeOptions = Record<string, EventTypeOptionsProps>;
+
+type LogLevelOptionsProps = {
+    icon: string;
+    color: string;
+    rowClass?: string;
+    text: string;
+};
+
+type LogLevelOptions = Record<string, LogLevelOptionsProps>;
```
Comment (Vorbert-Kruk) [inline app/widgets/common/utils/EventUtils.ts:18]: ### Optional

As
- `EventTypeOptionsProps` and `LogLevelOptionsProps` are not exported anywhere
- `EventTypeOptionsProps` and `LogLevelOptionsProps` are not being used outside of `EventTypeOptions` and `LogLevelOptions`
- Convention `{someName}Props` is often used for declaring component interfaces

WDYT about the following modification:

```suggestion
type EventTypeOptions = Record<
    string,
    {
        text: string;
        iconClass?: string;
        rowClass?: string;
    }
>;

type LogLevelOptions = Record<
    string,
    {
        icon: string;
        color: string;
        rowClass?: string;
        text: string;
    }
>;
```

? 🤔

File: widgets/eventsFilter/src/EventFilter.tsx [hunk 2]
```diff
@@ -50,16 +58,23 @@ function EventFilter({ toolbox }) {
...
-        return _.truncate(data.text, { length: 30 });
+    function renderLabel(data: DropdownItemProps) {
+        return _.truncate(data.text as string, { length: 30 });
     }
 
-    function handleInputChange(proxy, field) {
+    function handleInputChange(
+        _event: React.SyntheticEvent<HTMLElement, Event> | unknown,
+        field: InputOnChangeData | DateRangeInputOnChangeData | DropdownProps
+    ) {
```
Comment (Vorbert-Kruk) [inline widgets/eventsFilter/src/EventFilter.tsx:68]: We could simplify this code piece by doing the following imports at the beginning of the file:
```typescript
import type { DateRangeInputOnChangeData, DateRangeInputProps } from 'cloudify-ui-components';
import type { DropdownItemProps, DropdownProps, InputProps } from 'semantic-ui-react';
```

And modifying the highlighted code piece to be:
```typescript
    const handleInputChange: InputProps['onChange'] & DateRangeInputProps['onChange'] & DropdownProps['onChange'] = (
        _event,
        field
    ) => {
```

File: widgets/eventsFilter/src/EventFilter.tsx [hunk 3]
```diff
@@ -82,8 +97,9 @@ function EventFilter({ toolbox }) {
         debouncedContextUpdate(toolbox, updatedFields);
     }
 
-    function handleOptionAddition(e, { name, value }) {
-        setOptions({ ...options, [name]: [{ text: value, value }, ...options[name]] });
+    function handleOptionAddition(_event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) {
```
Comment (Vorbert-Kruk) [inline widgets/eventsFilter/src/EventFilter.tsx:100]: We can simplify this code piece, by doing

```suggestion
    const handleOptionAddition: DropdownProps['onAddItem'] = (_event, data) => {
```

**Review by Vorbert-Kruk — COMMENTED** [review]
Commented with some minor adjustment suggestions

Btw, PR description seems to be consisting of a plain PR description template 😄

File: app/widgets/common/utils/EventUtils.ts (hunk 1 above)
Comment (PatrycjaKubiczek) [reply app/widgets/common/utils/EventUtils.ts]: looks great, implemented it :)

File: widgets/eventsFilter/src/EventFilter.tsx (hunk 2 above)
Comment (PatrycjaKubiczek) [reply widgets/eventsFilter/src/EventFilter.tsx]: looks much, much better, thanks, implemted - https://github.com/cloudify-cosmo/cloudify-stage/pull/2446/commits/9e6dce8fa54ba7c5ba49d9b2c893ff32174a7677

File: widgets/eventsFilter/src/EventFilter.tsx (hunk 3 above)
Comment (PatrycjaKubiczek) [reply widgets/eventsFilter/src/EventFilter.tsx]: done - https://github.com/cloudify-cosmo/cloudify-stage/pull/2446/commits/9e6dce8fa54ba7c5ba49d9b2c893ff32174a7677 ✔️

**Review by Vorbert-Kruk — DISMISSED** [review]
LGTM 🎉 
Thanks for addressing all comments! 🚀

File: widgets/eventsFilter/src/EventFilter.tsx [hunk 4]
```diff
@@ -28,15 +33,18 @@ const defaultLogLevelOptions = _.map(_.keys(EventUtils.logLevelOptions), log =>
     value: log
 }));
 
-function isDirty(fields) {
-    return !_.every(initialFields, (value, name) => fields[name] === initialFields[name]);
+function isDirty(fields: { [x: string]: any }) {
```
Comment (qooban) [inline widgets/eventsFilter/src/EventFilter.tsx:36]: How about adding proper type to `fields` state variable? 
Won't it help in being able to reuse it here (proper type for function argument) as well as in other places (e.g. `updatedFields` variable)?

File: widgets/eventsFilter/src/EventFilter.tsx [hunk 5]
```diff
@@ -28,15 +33,18 @@ const defaultLogLevelOptions = _.map(_.keys(EventUtils.logLevelOptions), log =>
...
+    return !_.every(
```
Comment (qooban) [inline widgets/eventsFilter/src/EventFilter.tsx:37]: Optional:

We can remove `_.` uses.

Comment (PatrycjaKubiczek) [reply widgets/eventsFilter/src/EventFilter.tsx]: remove it - https://github.com/cloudify-cosmo/cloudify-stage/pull/2446/commits/66a972ed851da7e1c026fdaacecedebdc06b41d3 ✔️

File: widgets/eventsFilter/src/EventFilter.tsx (hunk 4 above)
Comment (PatrycjaKubiczek) [reply widgets/eventsFilter/src/EventFilter.tsx]: added typings for Fields: https://github.com/cloudify-cosmo/cloudify-stage/pull/2446/commits/bdce7e8a695f9fc1e146cabd838a572aca6ac596 ✔️

File: widgets/eventsFilter/src/EventFilter.tsx [hunk 6]
```diff
@@ -1,46 +1,67 @@
...
+interface Fields {
+    eventType: string[];
+    timeRange: DateRange;
+    timeStart: string | moment.Moment;
+    timeEnd: string | moment.Moment;
+    type: string;
+    messageText: string;
+    operationText: string;
+    logLevel: string[];
+    name: string;
```
Comment (qooban) [inline widgets/eventsFilter/src/EventFilter.tsx:19]: Are you sure there's `name` key in fields object? 🤔

Comment (PatrycjaKubiczek) [reply widgets/eventsFilter/src/EventFilter.tsx]: ah thanks, forget to remove that 🤦 https://github.com/cloudify-cosmo/cloudify-stage/pull/2446/commits/c7f06537df0119acb7aa3c7f3789375bcc0f21d2 done ✔️

**Review by qooban — APPROVED** [review]

**Pull request merged**